api/client/build.go

```diff
@@ -131,10 +130,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		return fmt.Errorf("cannot canonicalize dockerfile path %s: %v", relDockerfile, err)
 	}
 
-	var includes = []string{"."}
+	var (
+		includes = []string{"."}
+		excludes []string
+	)
 
-	excludes, err := utils.ReadDockerIgnore(path.Join(contextDir, ".dockerignore"))
-	if err != nil {
+	f, err := os.Open(filepath.Join(contextDir, ".dockerignore"))
```

Comment:
You are right that it is not strictly needed for a client-side builder, however if in the future we want to support builder plugins, it makes sense to abstract the context to allow for more possibilities. I'm also fine changing the Context interface to just handle paths instead of readers, it just seemed more appropriate like this after talking with others.